integration-cli/docker_cli_pull_test.go

```diff
@@ -150,3 +150,34 @@ func (s *DockerSuite) TestPullScratchNotAllowed(c *check.C) {
 		c.Fatalf("unexpected output pulling scratch: %s", out)
 	}
 }
+
+// pulling an image with --all-tags=true
+func (s *DockerSuite) TestPullImageWithAllTagFromCentralRegistry(c *check.C) {
+	testRequires(c, Network)
+	pullCmd := exec.Command(dockerBinary, "pull", "busybox")
+	if out, _, err := runCommandWithOutput(pullCmd); err != nil {
+		c.Fatalf("pulling the busybox image from the registry has failed: %s, %v", out, err)
+	}
+
+	ImageCmd := exec.Command(dockerBinary, "images", "busybox")
+	outImageCmd, exitCode, err := runCommandWithOutput(ImageCmd)
+
+	if err != nil || exitCode != 0 {
+		c.Fatalf("failed to run docker images busybox: %s, %v", outImageCmd, err)
+	}
+
+	pullAllTagCmd := exec.Command(dockerBinary, "pull", "--all-tags=true", "busybox")
+	if out, _, err := runCommandWithOutput(pullAllTagCmd); err != nil {
+		c.Fatalf("pulling the busybox image with all tags from the registry has failed: %s, %v", out, err)
+	}
+
+	outImageAllTagCmd, exitCode, err := runCommandWithOutput(ImageCmd)
+	if err != nil || exitCode != 0 {
+		c.Fatalf("failed to run docker images busybox: %s, %v", outImageAllTagCmd, err)
+	}
+
+	if strings.Count(outImageCmd, "busybox") >= strings.Count(outImageAllTagCmd, "busybox") {
+		c.Fatalf("Pulling with all tags should get more images")
+	}
```

Comment:
It might be good to run the 2nd command again but use "-a" instead of "--all-tags=true" just to make sure it has the same results.